src/crystal/system/win32/library_archive.cr

```diff
@@ -0,0 +1,85 @@
+# This module provides functions for operating with Windows library archives.
+# Implementation based on: https://docs.microsoft.com/en-us/windows/win32/debug/pe-format
+module Crystal::System::LibraryArchive
+  # Returns the list of DLL filenames imported by the .lib archive at the given
+  # *path*.
+  def self.imported_dlls(path : ::Path | String) : Set(String)
+    return Set(String).new unless ::File.file?(path)
+    ::File.open(path, "r") do |f|
+      reader = COFFReader.new(f)
+      reader.run
+      reader.dlls
+    end
+  end
+
+  # Minimal implementation of a Microsoft COFF archive reader. All unused fields
+  # are ignored.
+  private struct COFFReader
+    getter dlls = Set(String).new
```

Comment:
Why did you choose `Set` for this? There is an *order* to the filenames so I would expect to preserve that. We can still remove duplicates, of course.

Reply:
`Set` preserves insertion order, so it would be the same as using an `Array` and calling `#uniq!`, wouldn't it? Or is there some other order that should be observed here?

Reply:
It's not incidental. The concept of a set guarantees no order. The specific implementation of does. We are not going to change that, ever, as it would be a breaking change. Using a Set is fine here.

Reply:
Another way to say this: I really like how Set and Hash in Ruby preserve the order. That's why it was done in the same way in Crystal. It's a property of Ruby and Crystal Hash and Set and I'd like to keep it like that. It's efficient, and it also gives a guarantee about the order. There's absolutely no reason to change this, or to hide the fact that this works the way it is.